internal/app/vodtester/vodtester_app.go

```diff
@@ -144,11 +145,107 @@ func (vt *vodTester) Start(fileUrl string, taskPollDuration time.Duration) (int,
 		}
 	}
 
+	filePath := "/tmp/uploadtester.mp4"
+	resp, err := http.Get(fileUrl)
+
+	if err != nil {
+		glog.Errorf("Error downloading fileUrl=%s err=%v", fileUrl, err)
+	}
+	defer resp.Body.Close()
+
+	out, err := os.Create(filePath)
+	if err != nil {
+		glog.Errorf("Error creating file filePath=%s err=%v", filePath, err)
+	}
+	defer out.Close()
+
+	assetName = fmt.Sprintf("vod_test_asset_%s_%s", hostName, time.Now().Format("2006-01-02T15:04:05Z07:00"))
+	requestUpload, err := vt.lapi.RequestUpload(assetName)
+
+	if err != nil {
+		glog.Errorf("Error requesting upload for assetName=%s err=%v", assetName, err)
+		return 249, fmt.Errorf("error requesting upload for assetName=%s: %w", assetName, err)
+	}
+
+	uploadEndpoint := requestUpload.Url
+	uploadAsset := requestUpload.Asset
+	uploadTask := api.Task{
+		ID: requestUpload.Task.ID,
+	}
+
+	glog.Infof("Uploading to endpoint=%s", uploadEndpoint)
+
+	err = vt.lapi.UploadAsset(uploadEndpoint, out)
+	if err != nil {
+		glog.Errorf("Error uploading file filePath=%s err=%v", filePath, err)
+		return 250, fmt.Errorf("error uploading for assetId=%s taskId=%s: %w", uploadAsset.ID, uploadTask.ID, err)
+	}
+
+	err = vt.checkAssetProcessing(taskPollDuration, uploadAsset, uploadTask)
+
+	if err != nil {
+		glog.Errorf("Error processing asset assetId=%s taskId=%s", uploadAsset.ID, uploadTask.ID)
+		return 251, err
+	}
+
+	assetName = fmt.Sprintf("vod_test_asset_%s_%s", hostName, time.Now().Format("2006-01-02T15:04:05Z07:00"))
+	requestUpload, err = vt.lapi.RequestUpload(assetName)
+
+	if err != nil {
+		glog.Errorf("Error requesting upload for assetName=%s err=%v", assetName, err)
+		return 252, fmt.Errorf("error requesting upload for assetName=%s: %w", assetName, err)
+	}
+
+	tusUploadEndpoint := requestUpload.TusEndpoint
+	uploadAsset = requestUpload.Asset
+	uploadTask = api.Task{
+		ID: requestUpload.Task.ID,
+	}
+
+	err = vt.lapi.ResumableUpload(tusUploadEndpoint, filePath)
+
+	if err != nil {
+		glog.Errorf("Error resumable uploading file filePath=%s err=%v", filePath, err)
+		return 253, fmt.Errorf("error resumable uploading for assetId=%s taskId=%s: %w", uploadAsset.ID, uploadTask.ID, err)
+	}
+
+	err = vt.checkAssetProcessing(taskPollDuration, uploadAsset, uploadTask)
+
+	if err != nil {
+		glog.Errorf("Error processing asset assetId=%s taskId=%s", uploadAsset.ID, uploadTask.ID)
+		return 254, err
+	}
+
 	glog.Info("Done VOD Test")
 
 	return 0, nil
 }
 
+func (vt *vodTester) checkAssetProcessing(taskPollDuration time.Duration, uploadAsset api.Asset, uploadTask api.Task) error {
```

Comment:
Hmm, even better would be implementing a helper that works on top of a `Task` instead of an `Asset`, and then reuse it on the regular `import` (upload with URL) tests above as well. WDYT?

Reply:
Sounds good! I moved everything into a `checkTaskProcessing` and removed this function